v8env_test/proxy.spec.js

```diff
@@ -3,12 +3,22 @@ import { expect } from 'chai'
 import * as proxy from '@fly/proxy'
 
 const origin = "https://fly.io/proxy/"
-const req = new Request("https://wat.com/path/to/thing", { headers: { "host": "wut.com" } })
+const req = new Request("https://wat.com/path/to/thing", { headers: { "host": "wat.com" } })
+
 describe("proxy", () => {
   it('includes host header and base path properly', () => {
     const breq = proxy.buildProxyRequest(origin, {}, req)
     const url = new URL(breq.url)
+    expect(breq.headers.get("host")).to.eq("fly.io")
+    expect(url.hostname).to.eq("fly.io")
+    expect(url.pathname).to.eq("/proxy/path/to/thing")
+  })
+
+  it('includes explicit host header if different than origin', () => {
+    const breq = proxy.buildProxyRequest(origin, { headers: { "host": "wut.com" } }, req)
```

Comment:
Make sure that if an explicit `host` is given that it takes precedence over a naive rewrite.